x-pack/legacy/plugins/epm/server/datasources/create.ts

```diff
@@ -38,6 +39,14 @@ export async function createDatasource(options: {
   await baseSetup(callCluster);
   const pkg = await Registry.fetchInfo(pkgkey);
 
+  // Collect the config for each dataset
+  const configs: string[] = [];
+  if (pkg.datasets) {
```

Comment:
`pkg.datasets` refers to all the datasets that come with the package whereas `datasets` are the ones the user selected.